src/main/java/me/gallowsdove/foxymachines/listeners/SwordListener.java

```diff
@@ -19,111 +23,53 @@
-                            } else if (result < 60) {
-                                humanoid.addPotionEffect(new PotionEffect(PotionEffectType.WITHER, 80, 2, false, false));
-                            } else if (result < 80) {
-                                humanoid.damage(e.getDamage() / 2);
-                            } else {
-                                humanoid.addPotionEffect(new PotionEffect(PotionEffectType.CONFUSION, 150, 2, false, false));
-                            }
-                        }
-                    } else if (SlimefunUtils.isItemSimilar(item, Items.CELESTIAL_SWORD, false, false)) {
-
-                        Utils.dealDamageBypassingArmor(entity, (e.getDamage() - e.getFinalDamage()) * 0.16);
-
-                        if (random.nextInt(100) < 15) {
-                            entity.getWorld().strikeLightningEffect(entity.getLocation());
-                            entity.damage(8);
-                        }
-
-                        entity.addPotionEffect(new PotionEffect(PotionEffectType.GLOWING, 80, 0, false, false));
-
-                    } else if (SlimefunUtils.isItemSimilar(item, Items.ELUCIDATOR, false, false)) {
-
-                        e.setDamage(e.getDamage() * 2);
-
-                        Utils.dealDamageBypassingArmor(entity, (e.getDamage() - e.getFinalDamage()) * 0.064);
+    private void onDamage(EntityDamageByEntityEvent event) {
+        // If it's not a possible cause
+        if (event.getCause() != EntityDamageEvent.DamageCause.ENTITY_ATTACK && event.getCause() != EntityDamageEvent.DamageCause.ENTITY_SWEEP_ATTACK) {
+            return;
+        }
 
-                        double health = humanoid.getHealth() + 1.5D;
-                        double maxHealth = humanoid.getAttribute(Attribute.GENERIC_MAX_HEALTH).getValue();
+        // If the attacker is not a HumanEntity, someone who can't use the item, return
+        // Or if the attacked entity is not a living entity, someone who can't be attacked, return
+        if (!(event.getDamager() instanceof HumanEntity humanoid) || !(event.getEntity() instanceof LivingEntity entity)) {
+            return;
+        }
 
-                        if (health > maxHealth) {
-                            humanoid.setHealth(maxHealth);
-                            if (humanoid.getAbsorptionAmount() < 12) {
-                                humanoid.setAbsorptionAmount(Math.min(humanoid.getAbsorptionAmount() + (health - maxHealth) / 2, 12));
-                            }
-                        } else {
-                            humanoid.setHealth(health);
-                        }
-                    }
-                }
-            }
+        ThreadLocalRandom random = ThreadLocalRandom.current();
+        ItemStack item = humanoid.getInventory().getItemInMainHand();
+        if (SlimefunItem.getByItem(item) instanceof OnHitWeapon onHitWeapon) {
+            onHitWeapon.onHit(random, event, humanoid, entity);
         }
```

Comment:
Is there a reason why ThreadLocalRandom is passed as a parameter? I find it a quite weird design choice

Reply:
Yeah I can remove that, I wasn't sure of how I felt doing it when I did it in the first place it was just saving one line for them all 😅